pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -549,34 +550,58 @@ public final synchronized void readEntriesComplete(List<Entry> entries, Object c
             log.debug("[{}] Distributing {} messages to {} consumers", name, entries.size(), consumerList.size());
         }
 
+       sendMessagesToConsumers(readType, entries);
+    }
+
+    protected final synchronized void sendMessagesToConsumers(ReadType readType, List<Entry> entries) {
         // dispatch messages to a separate thread, but still in order for this subscription
```

Comment:
There is a lot of duplicate code in the method `sendMessagesToConsumers`, which I think should be optimized